Dalamud/Plugin/Internal/PluginManager.cs

```diff
@@ -595,82 +601,88 @@ public List<PluginUpdateStatus> UpdatePlugins(bool dryRun = false)
-                }
-                else
-                {
-                    // Unload if loaded
-                    if (plugin.State == PluginState.Loaded || plugin.State == PluginState.LoadError)
-                    {
-                        try
-                        {
-                            plugin.Unload();
-                        }
-                        catch (Exception ex)
-                        {
-                            Log.Error(ex, "Error during unload (update)");
-                            continue;
-                        }
-                    }
+            return updatedList;
+        }
 
-                    try
-                    {
-                        plugin.Disable();
-                        this.installedPlugins.Remove(plugin);
-                        listChanged = true;
-                    }
-                    catch (Exception ex)
-                    {
-                        Log.Error(ex, "Error during disable (update)");
-                        continue;
-                    }
+        /// <summary>
+        /// Update a single plugin, provided a valid <see cref="AvailablePluginUpdate"/>.
+        /// </summary>
+        /// <param name="metadata">The available plugin update.</param>
+        /// <param name="dryRun">Whether or not to actually perform the update, or just indicate success.</param>
+        /// <returns>The status of the update.</returns>
+        [CanBeNull]
+        public PluginUpdateStatus UpdateSinglePlugin(AvailablePluginUpdate metadata, bool dryRun)
+        {
+            var plugin = metadata.InstalledPlugin;
+
+            // Can't update that!
+            if (plugin is LocalDevPlugin)
+                return null;
+
+            var updateStatus = new PluginUpdateStatus
+            {
+                InternalName = plugin.Manifest.InternalName,
+                Name = plugin.Manifest.Name,
+                Version = metadata.UseTesting ? metadata.UpdateManifest.TestingAssemblyVersion : metadata.UpdateManifest.AssemblyVersion,
+            };
 
+            if (dryRun)
+            {
+                updateStatus.WasUpdated = true;
+            }
+            else
+            {
+                updateStatus.WasUpdated = true;
+
+                // Unload if loaded
+                if (plugin.State == PluginState.Loaded || plugin.State == PluginState.LoadError)
+                {
                     try
                     {
-                        this.InstallPlugin(metadata.UpdateManifest, metadata.UseTesting, PluginLoadReason.Update);
-                        listChanged = true;
+                        plugin.Unload();
                     }
                     catch (Exception ex)
                     {
-                        Log.Error(ex, "Error during install (update)");
-                        continue;
+                        Log.Error(ex, "Error during unload (update)");
+                        updateStatus.WasUpdated = false;
```

Comment:
Should return if there was an error on unloading. The plugin state will turn to UnloadError and the next operation will throw InvalidOperation